src/main/scala/path/Path.scala

```diff
@@ -5,33 +5,41 @@ import com.originate.scalypher.action.ReturnAll
 import com.originate.scalypher.action.ReturnDistinct
 import com.originate.scalypher.action.ReturnReference
 import com.originate.scalypher.Query
+import com.originate.scalypher.MatchQuery
+import com.originate.scalypher.MergeQuery
+import com.originate.scalypher.CreateQuery
 import com.originate.scalypher.types.Referenceable
 import com.originate.scalypher.types.ReferenceableMap
-import com.originate.scalypher.where.ReferenceType
+import com.originate.scalypher.where.Reference
 import com.originate.scalypher.where.Where
 
 case class PathWithWhere(path: Path, where: Where) {
```

Comment:
@chrizkim this is already halfway handled, i'll update as needed later though... `Action`s take `Reference`s, one kind of reference is `ReferenceWithProperty` (which you can create with the `property` method on referenceables):
```scala
val n = AnyNode()
val r = n property "thing"
val action = ReturnReference(r)
action.toQuery(Map(n -> "a")) // "RETURN a.thing"
```
the part that needs updating is that a query's return columns iterates over referenceables and not references (the latter includes the property if it's there). i'll save that for another PR.